### PR TITLE
Fix limit checking in QT figureoptions

### LINKED ISSUE #10496: exception after clicking ok when editing axis curve info on 3d plot
### Bug report

**Bug summary**

When plotting a 3d wireframe and opening the "Edit axis, curve and image properties" window, matplotlib will throw a ValueError exception after clicking "OK".

**Code for reproduction**

```python
import matplotlib.pyplot as plt
import numpy as np
from mpl_toolkits.mplot3d import Axes3D
from scipy.stats import norm

x = np.linspace(0,1,1000)
y = np.linspace(0,1,1000)
xx, yy = np.meshgrid(x,y)
fig = plt.figure()
ax = fig.add_subplot(111, projection='3d')
ax.plot_wireframe(xx,yy,norm.pdf(xx,0.5)*norm.pdf(yy,0.5))
plt.show()
```

Then click the "Edit axis, curve and image properties" button and then click "OK".

**Actual outcome**

When clicking "OK" I get the following exception:

```
$ python test.py 
Traceback (most recent call last):
  File "/home/tlatorre/.conda/envs/noisyopt/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 681, in edit_parameters
    figureoptions.figure_edit(axes, self)
  File "/home/tlatorre/.conda/envs/noisyopt/lib/python2.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py", line 262, in figure_edit
    apply_callback(data)
  File "/home/tlatorre/.conda/envs/noisyopt/lib/python2.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py", line 255, in apply_callback
    if not (axes.get_xlim() == orig_xlim and axes.get_ylim() == orig_ylim):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
Aborted
```

**Matplotlib version**

  * Operating system: Scientific Linux 6
  * Matplotlib version: 2.1.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 2.7.14

Matplotlib was installed using the anaconda installer.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #10496

### CODE REVIEW

**Comment by anntzer** [comment]
While this fix seems reasonable, I would argue that the correct thing to do is actually to make get_xlim() (and friends) also return tuples for 3d axes (similarly to the 2d case).

**Comment by dstansby** [comment]
Wouldn't that technically be changing the API (so probably not just a small bugfix). I'd be happy to do that change for 3.0, but maybe not for 2.2 now the RC is out?

**Comment by anntzer** [comment]
I would argue that making mplot3d behave more like regular axes by returning a tuple rather than an array is closer to a bugfix than an API change.

File: lib/matplotlib/backends/qt_editor/figureoptions.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def apply_callback(data):
         # Redraw
         figure = axes.get_figure()
         figure.canvas.draw()
-        if not (axes.get_xlim() == orig_xlim and axes.get_ylim() == orig_ylim):
+        if not ((axes.get_xlim() == orig_xlim).all() and
```
Comment (tacaswell) [inline lib/matplotlib/backends/qt_editor/figureoptions.py]: Won't this now break in the 2D case?

Comment (dstansby) [reply lib/matplotlib/backends/qt_editor/figureoptions.py]: 🤦‍♂️ oh yeah

**Comment by dstansby** [comment]
Closing in favour of #10554

**Pull request closed without merging**